PubHub/PubHub.Common/Services/BookService.cs

```diff
@@ -60,19 +58,20 @@ public async Task<List<BookInfoModel>> GetBooks(BookQuery queryOptions)
                 return null!;
             }
         }
+
         /// <summary>
         /// Calls the API end point for retrieving <see cref="BookInfoModel">, to use in the client applications.
         /// </summary>
         /// <param name="bookId">Id of the book we want information about.</param>
         /// <returns>A <see cref="BookInfoModel"/> with a book's information.</returns>
-        /// <exception cref="ArgumentOutOfRangeException"></exception>
+        /// <exception cref="ArgumentException"></exception>
         /// <exception cref="NullReferenceException"></exception>
-        public async Task<BookInfoModel?> GetBook(int bookId)
+        public async Task<BookInfoModel?> GetBook(Guid bookId)
         {
             try
             {
-                if (bookId <= 0)
-                    throw new ArgumentOutOfRangeException($"The book Id wasn't a valid Id: {bookId}");
+                if (bookId == Guid.Empty)
```

Comment:
Under `PubHub.Common` there's a constant collection called `INtegrityConstants` this collection contains a constant called `INVALID_ENTITY_ID`. You should use that instead (: